### PR TITLE
Exposing the test/mock package

### PR DESCRIPTION
**Pull request by AnomalRoil**
More specifically `chain.Beacon` doesn't have the same tags for JSON formatting as `client.RandomData` did and so it wasn't being marshaled as it used to when we used the `asRD` and the `RandomData` struct in our client code for the HTTP server.

And exposing the test/mock package, since these are needed in our drand-cli repo.

Also updating our `x/net` packages following its recent security patches.

### CODE REVIEW

File: demo/lib/orchestrator.go [hunk 1]
```diff
@@ -363,6 +363,12 @@ func (e *Orchestrator) checkBeaconNodes(nodes []node.Node, group string, tryCurl
 					panic("[-] Inconsistent round from curl vs CLI")
 				} else if !bytes.Equal(r.GetSignature(), pubRand.GetSignature()) {
 					fmt.Printf("curl output: %s\n", out)
+					if !strings.Contains(string(out), "round") ||
+						!strings.Contains(string(out), "randomness") ||
+						!strings.Contains(string(out), "signature") ||
+						!strings.Contains(string(out), "previous_signature") {
```
Comment (CluEleSsUK) [inline demo/lib/orchestrator.go:369]: won't this fail for unchained beacons?

File: handler/http/server_test.go [hunk 2]
```diff
@@ -53,24 +53,33 @@ func validateEndpoint(endpoint string, round float64) error {
...
-	if err := json.NewDecoder(resp.Body).Decode(&body); err != nil {
+	if err := json.NewDecoder(respBody).Decode(&body); err != nil {
 		return err
 	}
-	if err := resp.Body.Close(); err != nil {
-		return err
+	if len(body) != 4 && len(body) != 3 {
+		return fmt.Errorf("beacon formatting expects 3 or 4 fields for the beacon")
+	}
+	if rep, ok := body["round"].(float64); !ok || rep != round {
```
Comment (CluEleSsUK) [inline handler/http/server_test.go:67]: though a float in js land, it should never seem like a float by the time it reaches here (I appreciate this is just test code though)

File: demo/lib/orchestrator.go (hunk 1 above)
Comment (AnomalRoil) [reply demo/lib/orchestrator.go:369]: Good point

File: handler/http/server_test.go [hunk 3]
```diff
@@ -53,24 +53,33 @@ func validateEndpoint(endpoint string, round float64) error {
 		return fmt.Errorf("unexpected status: %v", resp.StatusCode)
 	}
 
+	return validateBodyFormat(resp.Body, round)
+}
+
+func validateBodyFormat(respBody io.Reader, round float64) error {
 	body := make(map[string]interface{})
```
[the remark this answers is not in the document]
Comment (AnomalRoil) [reply handler/http/server_test.go:60]: This is test code to catch such regressions in the future

**Review by CluEleSsUK — APPROVED** [review]

**Pull request merged**